c10/core/DispatchKey.h

```diff
@@ -214,6 +215,7 @@ enum class DispatchKey : uint8_t {
   AutogradXLA,
   AutogradXPU,
   AutogradMLC,
+  AutogradHPU,
```

Comment:
Just want to confirm that Habana is in fact overriding some autograd implementations with custom, Habana-specialized autograd.

Reply:
we discussed this morning, Sujoy mentioned an example of gelu where they store an extra forward value for saving for bwd, and that doing it via autograd override was easier than doing it in their backend due to their lazy-tensor capturing fragmented graphs and having to piece things back together across fragments.  

@sujoysaraswati is going to provide more examples of operators per @ailzhang request

Reply:
Few examples of autograd overwrite used in Habana -
1. Gelu: we save a tensor from the fwd op and pass it to the bwd of gelu. This helps accelerate the gelu backward kernel in Habana. To achieve this, we store the extra fwd output in the autograd context and pass it to the bwd op
2. isfinite: This is a compound op and PyTorch breaks it down into smaller ops. We wanted to prevent this from happening and pass the isfinite fwd and bwd op as a complete op to the existing isfinite Habana kernel and hence used autograd overwrite to achieve this.
3. matmul : Similar reasons as isfinite.